src/biogeochem/test/DustEmis_test/test_DustEmisZender2003.pf

```diff
@@ -0,0 +1,483 @@
+    call this%water_factory%setup_after_subgrid(snl = 3)
+    call this%soilstate_inst%Init(bounds)
+    do c = begc, endc
+       this%soilstate_inst%watsat_col(c,:) = 0.05_r8 * (c - bounds%begc + 1)
+    end do
+    call this%setupEnvironment( )
+    call this%water_factory%create_water_type(this%water_inst, watsat_col=this%soilstate_inst%watsat_col)
+    call this%atm2lnd_inst%InitForTesting(bounds, atm2lnd_params)
+    call this%canopystate_inst%SetNMLForTesting()
+    call this%canopystate_inst%Init(bounds)
+    call this%frictionvel_inst%InitForTesting(bounds)
+    urb_em(begl:endl) = 0.99_r8
+    call this%temperature_inst%Init(bounds,           &
+                               em_roof_lun=urb_em(begl:endl), &
+                               em_wall_lun=urb_em(begl:endl), &
+                               em_improad_lun=urb_em(begl:endl), &
+                               em_perroad_lun=urb_em(begl:endl), &
+                               is_simple_buildtemp=.true., is_prog_buildtemp=.false.)
+    deallocate (snow_depth_col )
+    deallocate( urb_em )
+  end subroutine setUp
+
+  !-----------------------------------------------------------------------
+
+  subroutine tearDown(this)
+    class(TestDustEmisZender2003), intent(inout) :: this
+
+    call this%dust_emis%Clean()
+    call this%water_factory%teardown(this%water_inst)
+    call SnowHydrologyClean()
+    call unittest_subgrid_teardown()
+    call this%atm2lnd_inst%Clean()
+    call clm_varcon_clean()
+    deallocate( this%filter_nolakep )
+  end subroutine tearDown
+
+  !-----------------------------------------------------------------------
+
+  subroutine setupEnvironment(this)
+    !
+    ! !DESCRIPTION:
+    ! Sets up the external environment used by Dust emissions - i.e., things accessed via
+    ! 'use' statements.
+    !
+    ! Assumes nlevgrnd and nlevsoi have been set, and that all necessary subgrid setup has
+    ! been completed.
+    !
+    use ColumnType, only : col
+    use GridcellType, only : grc
+    class(TestDustEmisZender2003), intent(in) :: this
+    !
+    integer :: c,j
+    real(r8), parameter :: clay = 10.0_r8
+
+    !-----------------------------------------------------------------------
+    col%dz(:,1:nlevgrnd) = 1.0_r8
+    do j = 1, nlevgrnd
+       do c = bounds%begc, bounds%endc
+          col%z(c,j) = sum(col%dz(c,1:j-1)) + 0.5_r8*col%dz(c,j)
+       end do
+    end do
+
+    do c = bounds%begc, bounds%endc
+       ! Setting nbedrock to nlevsoi means that none of the layers from 1:nlevsoi are
+       ! considered bedrock
+       col%nbedrock(c) = nlevsoi
+
+       this%soilstate_inst%gwc_thr_col(c) = ThresholdSoilMoistZender2003( clay )
+       this%soilstate_inst%mss_frc_cly_vld_col(c) = MassFracClay( clay )
+    end do
+
+
+    ! Use longitude along Greenich so don't have to calculate offsets for longitudes (that's calculated in clm_time_manager)
+    grc%londeg(:) = 0.0_r8
```

Comment:
I'm pretty sure neither of these two settings are really needed so I could remove them.